presto-main/src/main/java/io/prestosql/server/protocol/ExecutingStatementResource.java

```diff
@@ -106,6 +113,34 @@ public ExecutingStatementResource(
         this.blockEncodingSerde = requireNonNull(blockEncodingSerde, "blockEncodingSerde is null");
         this.responseExecutor = requireNonNull(responseExecutor, "responseExecutor is null");
         this.timeoutExecutor = requireNonNull(timeoutExecutor, "timeoutExecutor is null");
+
+        queryPurger.scheduleWithFixedDelay(
+                () -> {
+                    try {
+                        for (Entry<QueryId, Query> entry : queries.entrySet()) {
+                            // forget about this query if the query manager is no longer tracking it
+                            try {
+                                queryManager.getQueryInfo(entry.getKey());
```

Comment:
`#getQueryInfo` isn't free to call. Maybe `#getQueryState` would be cheaper and with the same effect of checking the query id validity? (or a dedicated method)